### ISSUE TITLE
Support for turret firing arcs

### ISSUE THREAD

**Issue by YeOldeEBM** [OpenRA/OpenRA#10855]
Request. Turrets currently only function with 360 degree rotation. Firing arcs would increase flexibility with unit designs.

**Comment by KOYK** [comment]
Yes please, i was about to post the same issue, this is highly needed.
